packages/form-js-editor/src/features/properties-panel/PropertiesPanel.js

```diff
@@ -83,6 +85,17 @@ export function PropertiesPanel(props) {
     );
   }, [providers, selectedFormField, editField]);
 
+  const formFields = getService('formFields');
+
+  const PropertiesPanelHeaderProvider = useMemo(
+    () =>
+      getPropertiesPanelHeaderProvider({
+        getDocumentationRef: propertiesPanelConfig.getDocumentationRef,
+        formFields,
+      }),
+    [formFields, propertiesPanelConfig],
+  );
```

Comment:
Was this memo really necessary? If you didn't see any noticeable performance issue without it we should remove to avoid memory leaks

Reply:
Nothing is necessary but not having this means we re-render the whole header every render cycle. Which isn't ideal. It won't lag, yet, but could cause problems down the line.

Basically, this header provider stuff is a config, and we need to keep the reference to the config stable, hence the memo.

Reply:
Just rerendering is not an issue, as long as rerendering doesn't take long. Also, not all rerenders are committed to the DOM which is usually the most expensive.

But overusing memoization might cause memory leaks as explained [here](https://schiener.io/2024-03-03/react-closures)